apps/omg_performance/test/omg_performance/integration/bazantine_event_test.exs

```diff
@@ -0,0 +1,145 @@
+
+  @moduletag :integration
+  @moduletag timeout: 70_000
+
+  @tag fixtures: [:watcher, :child_chain, :contract]
+  test "time response for asking for exit data", %{contract: %{contract_addr: contract}} do
+    dos_users = 10
+    ntx_to_send = 10
+    spenders = generate_users(2)
+
+    IO.puts("""
+    dos users: #{dos_users}
+    spenders: #{length(spenders)}
+    ntx_toxend: #{ntx_to_send}
+    exits per dos user: #{length(spenders) * ntx_to_send}
+    total exits: #{length(spenders) * ntx_to_send * dos_users}
+    """)
+
+    # TODO find another way to wait for data generations, maybe smoke_test_statistics
+    Performance.start_extended_perftest(ntx_to_send, spenders, contract)
+    Process.sleep(4000)
+
+    statistics =
+      Enum.map(1..dos_users, fn _ ->
+        Task.async(fn ->
+          stream_exit_data_blocks = generate_exit_info() |> Enum.map(&get_lazy_exit_data/1) |> Enum.to_list()
+          :timer.tc(fn -> stream_exit_data_blocks |> Enum.map(&Enum.to_list(&1)) end)
+        end)
+      end)
+      |> Enum.map(fn task ->
+        {time, exit_data_blocks} = Task.await(task, :infinity)
+
+        {time / 1_000,
+         exit_data_blocks
+         |> Enum.map(fn exits_utxo ->
+           Enum.reduce(exits_utxo, %{correct: 0, error: 0}, fn
+             %{"proof" => _}, stats -> Map.update(stats, :correct, 0, &(&1 + 1))
+             _, stats -> Map.update(stats, :error, 0, &(&1 + 1))
+           end)
+         end)
+         |> Enum.reduce(%{error: 0, correct: 0}, fn %{error: error, correct: correct}, statistics ->
+           Map.update(statistics, :error, 0, &(&1 + error))
+           |> Map.update(:correct, 0, &(&1 + correct))
+         end)}
+      end)
+
+    times = statistics |> Enum.map(&elem(&1, 0))
+    correct_exits = statistics |> Enum.map(&Map.get(elem(&1, 1), :correct)) |> Enum.sum()
+    error_exits = statistics |> Enum.map(&Map.get(elem(&1, 1), :error)) |> Enum.sum()
+
+    IO.puts("""
+    max dos user time: #{Enum.max(times)}
+    min dos user time: #{Enum.min(times)}
+    average dos user time: #{Enum.sum(times) / length(times)}
+    correct exits: #{correct_exits}
+    error exits: #{error_exits}
+    """)
+
+    assert error_exits == 0
+  end
+
+  defp generate_exit_info do
+    %{interval: interval, counter: length_block} = block_info()
+
+    Enum.shuffle(1..length_block)
+    |> Stream.map(&generate_exit_info_block(&1 * interval))
+  end
+
+  defp block_info do
+    {:ok, interval} = RootChain.get_child_block_interval()
+    {:ok, top_block} = RootChain.get_mined_child_block()
+    %{interval: interval, counter: trunc(top_block / interval)}
+  end
+
+  defp generate_exit_info_block(blknum) do
+    {:ok, {block_hash, _timestamp}} = RootChain.get_child_chain(blknum)
+    child_chain_url = Application.get_env(:omg_watcher, :child_chain_url)
+    {:ok, block} = Client.get_block(block_hash, child_chain_url)
+
+    Enum.shuffle(Enum.zip(block.transactions, Stream.iterate(0, &(&1 + 1))))
```

Comment:
There is `Enum.with_index(...)` instead of `Enum.zip(block.transactions, Stream.iterate(...)))`
Also you can use `Stream` instead of `Enum` as you don't  need an intermediate result